docs/framework/docker/console.md

```diff
@@ -0,0 +1,139 @@
+---
+# Running console applications in Windows containers
+
+Console applications are used for many purposes; from simple querying of a status to long running document image processing tasks. In either case, the ability to start up and scale these applications are met with limitations of hardware acquisitions, startup times or running multiple instances.
+
+Moving your console applications to use Docker and Windows containers allows for starting these applications from clean state, perform the operation and then shutdown cleanly. This topic will show the steps needed to move a console application to a Windows based container and start it using a PowerShell script.
+
+The sample console application is a simple example which takes an argument, a question in this case, and returns a random answer. This could take a `customer_id` and process their taxes, or create a thumbnail for an `image_url` argument.
+
+In addition to the answer, the `Environment.MachineName` has been added to the end of the answer to show when running the application locally, your local machine name should be returned and when running in a Windows Container; the container session id is returned.
+
+The completed example is available in the [dotnet/core-docs repository on GitHub](https://github.com/dotnet/core-docs/tree/master/samples/framework/docker/ConsoleRandomAnswerGenerator).
+
+You need to be familiar with some Docker terms before you begin working
+on moving your application to a container.
+
+> A **Docker image** is a read-only template that defines the environment
+> for a running container, including the OS, system components, and application(s).
+
+One important feature of Docker images is that images are composed from a
+base image. Each new image adds a small set of features to an existing
+image. 
+
+> A **Docker container** is a running instance of an image. 
+
+You scale an application by running the same image in many containers.
+Conceptually, this is similar to running the same application in multiple
+hosts.
+
+You can learn more about the Docker architecture by reading the 
+[Docker Overview](https://docs.docker.com/engine/understanding-docker/)
+on the Docker site. 
+
+## Getting started
+You'll need to have Docker for Windows, version 1.12 Beta 26 or higher to support Windows containers. By default, Docker enables Linux based containers; switch to Windows containers by right clicking the Docker icon in the system tray and select "Switch to Windows Containers...". Docker will run the process to change and a restart may be required.
+
+Moving your console application is a matter of a few steps.
+
+1. Build the application
+1. Creating a Dockerfile for the image
+1. Process to build and run the Docker container
+
+## Building the application
+Typically console applications are distributed through an installer, FTP, or File Share deployment. Here the assets need to be compiled and staged to a location that can be used when the Docker image is created.
```

Comment:
> Typically console applications are distributed through an installer, FTP, or File Share deployment. Here the assets need to be compiled and staged to a location that can be used when the Docker image is created.

Typically console applications are distributed through an installer, FTP, or File Share deployment. When deploying to a container, the assets need to be compiled and staged to a location that can be used when the Docker image is created.